### PR TITLE
Drop PSA's censor-codes/filter-codes

### LINKED ISSUE #1067: Simplify PSA option of warning censoring to 'all', list of codes to censor, and list of messages to censor via prefix test
I propose we refactor the PSA warning censoring options to one of these two values:
```yaml
censor_warnings:
  # Case 1: a string
  "all"

  # Case 2: an object where one of these keys is optional
  censor_by_code:
    - CodeName1
    - CodeName2
  
  censor_by_prefix:
    - "\"Data.Map\"'s `Semigroup` instance" 
```

### PR DESCRIPTION
**Pull request by JordanMartinez**
### Description of the change

Fixes #1067

### Checklist:

- [ ] Added the change to the "Unreleased" section of the changelog
- [ ] Added some example of the new feature to the `README`
- [ ] Added a test for the contribution (if applicable)

**P.S.**: the above checks are not compulsory to get a change merged, so you may skip them. However, taking care of them will result in less work for the maintainers and will be much appreciated 😊

### CODE REVIEW

File: src/Spago/Config.purs [hunk 1]
```diff
@@ -49,7 +48,7 @@ import Registry.PackageName as PackageName
 import Registry.Range as Range
 import Registry.Sha256 as Sha256
 import Registry.Version as Version
-import Spago.Core.Config as Core
```
Comment (JordanMartinez) [inline src/Spago/Config.purs:52]: Ugh. Let me fix this...

File: README.md [hunk 2]
```diff
@@ -1192,18 +1192,26 @@ workspace:
...
-    # Optional array of warning codes to show for files in the
-    # `.spago` directory. Only useful if `censor_library_warnings` is not 'none'
-    # and the codes here do not appear in `censor_library_codes`.
-    filter_library_codes:
-      - UnusedName
+    # Optional and can be one of three possible values
+    censor_library_warnings: 
+      # Value 1 (default): "none" - All warnings are printed
+      none
+
```
[the remark this answers is not in the document]
Comment (JordanMartinez) [reply README.md:1199]: Ah... Well... yes, I suppose that works :smile:

[the remark this answers is not in the document]
Comment (JordanMartinez) [reply README.md]: Does dropping the `NoWarnings` ctor and just wrapping it in `Maybe` and updating `shouldPrintWarning` work for ya?

**Comment by JordanMartinez** [comment]
CI now builds.

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (f-f) [reply README.md]: Both `Maybe NonEmptyArray` and `Array` work for me 😄

**Review by f-f — APPROVED** [review]

**Pull request merged**